packages/alfa-aria/src/roles/widgets/combobox.ts

```diff
@@ -0,0 +1,29 @@
+import { Element, getAttribute, Node } from "@siteimprove/alfa-dom";
+import { Role } from "../../types";
+import * as Attributes from "../../attributes";
+import { Select } from "../abstract";
+import { TextBox, ListBox, Tree, Grid } from "../widgets";
+import { Dialog } from "../window";
+
+/**
+ * @see https://www.w3.org/TR/wai-aria/#combobox
+ */
+export const Combobox: Role = {
+  name: "combobox",
+  inherits: [Select],
+  owned,
+  required: [Attributes.Controls, Attributes.Expanded],
+  supported: [
+    Attributes.Autocomplete,
+    Attributes.ReadOnly,
+    Attributes.Required
+  ],
+  label: { from: ["author"], required: true }
+};
+
+function owned(combobox: Element, context: Node): Array<Role | [Role, Role]> {
+  if (getAttribute(combobox, "expanded") === "true") {
+    return ([TextBox] && [ListBox]) || [Tree] || [Grid] || [Dialog];
```

Comment:
Something we need to talk about? 🙈

Reply:
Words cannot explain what I was thinking when writing this.